components/kyma-environment-broker/internal/orchestration/handlers/kyma_handler.go

```diff
@@ -74,6 +74,22 @@ func (h *kymaHandler) createOrchestration(w http.ResponseWriter, r *http.Request
 		return
 	}
 
+	// validate deprecated parameteter `maintenanceWindow`
+	err = h.ValidateDeprecatedParameters(params)
+	if err != nil {
+		h.log.Errorf("found deprecated value: %v", err)
+		httputil.WriteErrorResponse(w, http.StatusBadRequest, errors.Wrapf(err, "found deprecated value"))
+		return
+	}
+
+	// validate `schedule` field
+	err = h.ValidateScheduleParameter(&params)
+	if err != nil {
+		h.log.Errorf("found deprecated value: %v", err)
+		httputil.WriteErrorResponse(w, http.StatusBadRequest, errors.Wrapf(err, "found deprecated value"))
+		return
+	}
+
```

Comment:
I would extract and move these to the handlers.go as this conversion/validation should also be used by cluster upgrade handler.

Reply:
Moved the validation methods to `handlers.go` and added validation to `cluster_handler.go`.